src/app/shared/issue-tables/search-filter.ts

```diff
@@ -0,0 +1,43 @@
+import { IssueService } from '../../core/services/issue.service';
+import { Issue } from '../../core/models/issue.model';
+import { TABLE_COLUMNS } from './issue-tables.component';
+
+export function applySearchFilter(filter: string, displayedColumn: string[], issueService: IssueService, data: Issue[]): Issue[] {
+    const searchKey = filter.toLowerCase();
+    const result = data.slice().filter((issue: Issue) => {
+      for (const column of displayedColumn) {
+        switch (column) {
+          case TABLE_COLUMNS.ASSIGNEE:
+            for (const assignee of issue.assignees) {
```

Comment:
For further readability, can we move the search logic for each column into a separate method?

e.g. a method `matchesAssignee` that searches through the issues' assignees; a `matchesDuplicatedIssue` method that does the search through the issues' duplicated methods, etc

No need to test each method individually. The current tests will cover them.